generation/new_client_hermetic_build/new-client.py

```diff
@@ -0,0 +1,253 @@
+    "com.google.cloud:google-cloud-<api_shortname>",
+)
+@click.option(
+    "--release-level",
+    type=click.Choice(["stable", "preview"]),
+    default="preview",
+    show_default=True,
+    help="A label that appears in repo-metadata.json. The first library "
+         "generation is always 'preview'."
+)
+@click.option(
+    "--api-id",
+    type=str,
+    help="The value of the apiid parameter used in README.md It has link to "
+         "https://console.cloud.google.com/flows/enableapi?apiid=<api_id>"
+)
+@click.option(
+    "--requires-billing",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="Based on this value, README.md explains whether billing setup is "
+         "needed or not."
+)
+@click.option(
+    "--cloud-api",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="If true, the artifact ID of the library is 'google-cloud-'; "
+         "otherwise 'google-'"
+)
+@click.option(
+    "--group-id",
+    type=str,
+    default="com.google.cloud",
+    show_default=True,
+    help="The group ID of the artifact when distribution name is not set"
+)
+@click.option(
+    "--library-type",
+    type=str,
+    default="GAPIC_AUTO",
+    show_default=True,
+    help="A label that appears in repo-metadata.json to tell how the library is "
+         "maintained or generated"
+)
+def add_new_library(
+    api_shortname,
+    name_pretty,
+    proto_path,
+    product_docs,
+    rest_docs,
+    rpc_docs,
+    api_description,
+    transport,
+    destination_name,
+    distribution_name,
+    release_level,
+    api_id,
+    requires_billing,
+    cloud_api,
+    group_id,
+    library_type,
+):
+    cloud_prefix = "cloud-" if cloud_api else ""
+    output_name = destination_name if destination_name else api_shortname
+    if distribution_name is None:
+        distribution_name = f"{group_id}:google-{cloud_prefix}{output_name}"
+
+    distribution_name_short = re.split(r"[:\/]", distribution_name)[-1]
+
+    if api_id is None:
+        api_id = f"{api_shortname}.googleapis.com"
+
+    if not product_docs.startswith("https"):
+        sys.exit(
+            f"product_docs must starts with 'https://' - actual value is {product_docs}"
+        )
+
+    client_documentation = f"https://cloud.google.com/java/docs/reference/{distribution_name_short}/latest/overview"
+
+    if destination_name is None:
+        destination_name = f"java-{api_shortname}"
+
+    if api_shortname == "":
+        sys.exit("api_shortname is empty")
+
+    path_to_yaml = os.path.join(script_dir, "..", "..", "generation_config.yaml")
+    with open(path_to_yaml, "r") as file_stream:
+        config = yaml.load(file_stream)
+
+    # confirm library doesn't exist both in the monorepo and config yaml
+    path_to_library = os.path.join(script_dir, "..", "..", destination_name)
+
+    for library in config["libraries"]:
+        if __compute_destination_name(library) == destination_name:
```

Comment:
We have a logic to check the uniqueness of a library name, why do we need similar logic here?

Reply:
[Found it](https://github.com/googleapis/sdk-platform-java/blob/4d3c06e361f18a851eca249ac34f357273bb0c51/library_generation/utilities.py#L262-L265) - I wasn't aware of this check. Thank you.